### PR TITLE
Issue 3009: Document fix streamcuts.md

### LINKED ISSUE #3009: Update StreamCut documentation.
**Problem description**
Update StreamCut documentation to include `io.pravega.client.stream.ReaderGroup#generateStreamCuts` 

**Problem location**
StreamCuts.md

**Suggestions for an improvement**
Add documentation.

### PR DESCRIPTION
**Pull request by Aparnarr**
Signed-off-by: Aparna <Aparna.Rr@emc.com>

**Change log description**  
 1. version number changed (refer last line in the document)
 2.  Update io.pravega.client.stream.ReaderGroup#generateStreamCuts (TBD)

**Purpose of the change**  
 Fixes #3009

### CODE REVIEW

**Comment by Aparnarr** [comment]
@shrids 
Should i replace in line 41??

**Comment by shrids** [comment]
Updates required 
-  Once PR #2980 is merged there will be an additional way to fetch `StreamCut`s using a `ReaderGroup` ref: [1]
-  With PR #2951 the method to obtain the HEAD and TAIL `StreamCut`s is via `StreamManager`. The current document refers to `BatchClient#getStreamInfo` This has been deprecated and needs to be removed from the documentation. ref [2] for details.


==================================================================
[1]
The two ways of fetching `StreamCut`s using `ReaderGroup` are 
i) Obtain `StreamCut`s using api ``` io.pravega.client.stream.ReaderGroup.getStreamCuts ```
This API returns a
```Map<Stream, StreamCut>``` which represents the last known position of the Readers for all the streams managed by the ReaderGroup.

ii) The newly added api in PR #2980 .

>  io.pravega.client.stream.ReaderGroup#generateStreamCuts`

`This api generates a StreamCut after co-ordinating with all the readers using io.pravega.client.state.StateSynchronizer. A StreamCut is generated by using the latest segment read offsets returned by the readers along with unassigned segments (if any). The configuration ReaderGroupConfig.getGroupRefreshTimeMillis() decides the maximum delay by which the readers return the latest read offsets of their assigned segments.`
`The StreamCut generated by this API can be used by the application as a reference to a position in the stream. This is guaranteed to be greater than or equal to the position of the readers at the point of invocation of the API.`

[2] The newly added api in PR #2951 is 
```io.pravega.client.admin.StreamManager#getStreamInfo```
This will provide the `HEAD` and `TAIL` StreamCut for a `Stream`.

File: documentation/src/docs/streamcuts.md [hunk 1]
```diff
@@ -91,4 +91,4 @@ fetch segments which reside between the given `startStreamCut` and `endStreamCut
 It must be noted that passing ```StreamCut.UNBOUNDED``` to `startStreamCut` and `endStreamCut` will result in using the current head of stream and the current tail of the stream, respectively.
 
 
-We have provided a simple yet illustrative example of using StreamCut [here](https://github.com/pravega/pravega-samples/tree/v0.3.2/pravega-client-examples/src/main/java/io/pravega/example/streamcuts).
```
Comment (shrids) [inline documentation/src/docs/streamcuts.md]: The new apis to obtain StreamCuts needs to be added to the documentation.

Comment (Aparnarr) [reply documentation/src/docs/streamcuts.md]: Addressed Your comments, Please check!

File: documentation/src/docs/streamcuts.md [hunk 2]
```diff
@@ -29,22 +29,22 @@ events can be continuously added to the stream and the `StreamCut` pointing to t
...
 ## StreamCut with Reader
 
-A ReaderGroup is a named collection of Readers that together, in parallel, read Events from a given Stream. Every
-Reader is always associated with a ReaderGroup. `StreamCut`(s) can be obtained from a ReaderGroup using the
-following API ```io.pravega.client.stream.ReaderGroup.getStreamCuts ```. This API returns a
-```Map<Stream, StreamCut>``` which represents the last known position of the Readers for all the streams managed by the ReaderGroup.
+A Reader Group is a named collection of Readers that together, in parallel, read Events from a given stream. Every
+Reader is always associated with a Reader Group. `StreamCut`(s) can be obtained from a Reader Group using the
+following API `io.pravega.client.stream.ReaderGroup.getStreamCuts `. This API returns a
+`Map<Stream, StreamCut>` which represents the **last known Position of the Readers** for all the streams managed by the Reader Group.
```
Comment (shrids) [inline documentation/src/docs/streamcuts.md]: Perhaps we could add the `generateStreamCuts()` api here. There are two ways to fetch `StreamCut`s pointing to the reader position.

File: documentation/src/docs/streamcuts.md [hunk 3]
```diff
@@ -65,30 +65,38 @@ ReaderGroupConfig.builder()
                 .build();
 ```
 
-The below API can be used to reset an existing ReaderGroup with a new ReaderGroup configuration instead creating a
-ReaderGroup.
+The below API can be used to **reset an existing Reader Group with a new Reader Group** configuration instead creating a
```
Comment (shrids) [inline documentation/src/docs/streamcuts.md]: We need not highlight here.

File: documentation/src/docs/streamcuts.md [hunk 4]
```diff
@@ -65,30 +65,38 @@ ReaderGroupConfig.builder()
...
 io.pravega.client.stream.ReaderGroup.resetReaderGroup(ReaderGroupConfig config)
 ```
+The API `io.pravega.client.stream.ReaderGroup#generateStreamCuts`, generates a `StreamCut` after co-ordinating with all the Readers using `io.pravega.client.state.StateSynchronizer`. A `StreamCut` is generated by using the latest Stream Segment read offsets returned by the Readers along with unassigned segments (if any). The configuration `ReaderGroupConfig.getGroupRefreshTimeMillis()` decides the maximum delay by which the Readers return the latest read offsets of their assigned segments.
+
+The `StreamCut` generated by this API can be used by the application as a reference to a Position in the stream. This is guaranteed to be greater than or equal to the position of the Readers at the point of invocation of the API.
+
 ## StreamCut with BatchClient
 
-`StreamCut` representing the current head and current tail of a stream can be obtained using below BatchClient API.
+`StreamCut` representing the current head and current tail of a stream can be obtained using below `BatchClient` API.
```
Comment (shrids) [inline documentation/src/docs/streamcuts.md]: This should be removed, as the api to fetch the current `TAIL and HEAD` of the stream is  on `StreamManager` and not `BatchClient`.

Perhaps we could add a section which talks about obtaining `HEAD` and `TAIL` streamcuts for a Stream. In the batch client section should contain how it is used as in line 88.

File: documentation/src/docs/streamcuts.md [hunk 5]
```diff
@@ -65,30 +65,38 @@ ReaderGroupConfig.builder()
...
+
+Enable `getStreamInfo(String scopeName, String streamName)` API on `StreamManager`. By enabling the `getStreamInfo()` API on `StreamManager` and deprecates the `getStreamInfo()` API inside the `BatchClient`. This ensure the `getStreamInfo()` API can be used outside the `BatchClient`.
```
Comment (shrids) [inline documentation/src/docs/streamcuts.md]: This needs to be removed/refactored...

File: documentation/src/docs/streamcuts.md (hunk 2 above)
Comment (Aparnarr) [reply documentation/src/docs/streamcuts.md]: done

File: documentation/src/docs/streamcuts.md (hunk 3 above)
Comment (Aparnarr) [reply documentation/src/docs/streamcuts.md]: done

File: documentation/src/docs/streamcuts.md [hunk 6]
```diff
@@ -65,30 +69,45 @@ ReaderGroupConfig.builder()
...
 ```
 /*
  * ReaderGroup API used to reset a ReaderGroup to a newer ReaderGroup configuration.
  */
 io.pravega.client.stream.ReaderGroup.resetReaderGroup(ReaderGroupConfig config)
+```
+
+## StreamCut with Stream Manager
+
+`StreamCut` representing the current head and current tail of a stream can be obtained using the `StreamManager` API. The `getStreamInfo(String scopeName, String streamName)` API on `StreamManager` ensures that the `getStreamInfo()` API can be used outside the `BatchClient`.
```
Comment (shrids) [inline documentation/src/docs/streamcuts.md]: Both the sentences could be combined . No need to mention the BatchClient.
something like the following...
```
`StreamCut`s representing the current head and current tail of the stream can be obtained by the `getStreamInfo(String scope, String streamName ) API on the the `StreamManager`
```

File: documentation/src/docs/streamcuts.md [hunk 7]
```diff
@@ -65,30 +68,45 @@ ReaderGroupConfig.builder()
...
+ * used to fetch the StreamCuts.
+ */
+CompletableFuture<StreamInfo> getStreamInfo(Stream stream);
+
 ```
 ## StreamCut with BatchClient
 
-`StreamCut` representing the current head and current tail of a stream can be obtained using below BatchClient API.
 ```
 /*
```
Comment (shrids) [inline documentation/src/docs/streamcuts.md]: please remove lines 96 to 104

**Pull request merged**